fiftyone/utils/openimages.py

```diff
@@ -0,0 +1,1325 @@
+                filtered_classes.append(c)
+            except:
+                missing_classes.append(c)
+        classes = filtered_classes
+        if missing_classes:
+            logger.info(
+                "The following are not available classes: %s\n\nSee available "
+                "classes with fouo.get_classes()\n" % ",".join(missing_classes)
+            )
+
+    attrs = []
+    attrs_map = {}
+    oi_attrs = []
+    if "relationships" in label_types:
+        # Map of attribute IDs to attribute names
+        attrs_map = _get_attrs_map(
+            dataset_dir=dataset_dir, scratch_dir=scratch_dir
+        )
+
+        all_attrs = sorted(list(attrs_map.values()))
+        dataset.info["attributes_map"] = attrs_map
+        dataset.info["attributes"] = all_attrs
+
+        if attrs == None:
+            oi_attrs = list(attrs_map.keys())
+            attrs = all_attrs
+
+        else:
+            attrs_map_rev = {v: k for k, v in attrs_map.items()}
+            missing_attrs = []
+            filtered_attrs = []
+            for a in attrs:
+                try:
+                    oi_attrs.append(attrs_map_rev[a])
+                    filtered_attrs.append(a)
+                except:
+                    missing_attrs.append(a)
+            attrs = filtered_attrs
+            if missing_attrs:
+                logger.info(
+                    "The following are not available attributes: %s\n\nSee "
+                    "available attributes with fouo.get_attributes()\n"
+                    % ",".join(missing_attrs)
+                )
+
+    seg_classes = []
+    if "segmentations" in label_types:
+        seg_classes = get_segmentation_classes(
+            dataset_dir=dataset_dir,
+            scratch_dir=scratch_dir,
+            classes_map=classes_map,
+        )
+
+        dataset.info["segmentation_classes"] = seg_classes
+
+    # Add class hierarchy to dataset.info, used in evaluation
+    hierarchy = _get_hierarchy(
+        dataset_dir=dataset_dir,
+        scratch_dir=scratch_dir,
+        classes_map=classes_map,
+    )
+    dataset.info["hierarchy"] = hierarchy
+
+    dataset = _load_open_images_split(
+        dataset,
+        label_types,
+        guarantee_all_types,
+        split_image_ids,
+        classes_map,
+        attrs_map,
+        oi_classes,
+        oi_attrs,
+        seg_classes,
+        dataset_dir,
+        scratch_dir,
+        split,
+        classes,
+        attrs,
+        max_samples,
+        num_workers,
+    )
+
+    # Export the labels of each split in FiftyOneDataset format
+    export_dir = os.path.join(dataset_dir, split)
+    logger.info("Writing annotations for %s split to disk" % split)
+    dataset.export(
+        export_dir, dataset_type=fot.FiftyOneDataset, export_media=False
+    )
+    num_samples = len(dataset)
+    dataset.delete()
+
+    return num_samples, all_classes
+
+
+def get_attributes(dataset_dir=None, scratch_dir=None):
```

Comment:
No need to provide the `scratch_dir` argument here, right? The way I see it, the cleanest way for this method to function is:

- Check if dataset is downloaded in default location (or `dataset_dir`, if provided)
- If relevant files exist, try to load attributes from metadata or scratch dir
- If we fail, temporarily download the attributes file, load the attributes, and then delete the temporary file

Note in the last case that I'm recommending to cleanup the file, since this mode corresponds to a user who just wants to use our code to get the classes without consenting to storing anything on disk. (you can use `etau.TempDir()` to do this cleanly)

I'd also recommend mentioning in the docstring that this method can be called in isolation without having the dataset downloaded.

This comment applies to the other public methods below.

Reply:
update: after running the method; since it does log where the file is downloaded, I'm okay with `get_attributes()` not cleaning up after itself automatically

Reply:
I kind of like it cleaning up after itself. There is always the option for you to just download a subset of the dataset if you don't want to have to download the files every time.